.changeset/selfish-numbers-sparkle.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector-class-pattern` selector message argument be trimmed
```

Comment:
```suggestion
Fixed: `selector-class-pattern` start positions
```

This is more in line with the other change log entries.

Reply:
Check the actual change again.
This is not about the reported ranges.
Granted the actual changeset is bad but yours is not accurate.

Reply:
```suggestion
Fixed: `selector-class-pattern` first message argument
```

Reply:
I don't think it is inaccurate.

The `endIndex` is the sum of the selector length and the start `index`.
This changes affects the selector length and thereby also the `endIndex`

You can also see how the `endIndex` is incorrect in the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AOFAPANAAlRgfAJYC2A5tsAO6FQAuAFvGgA7oC+ImIAZoQDYwAcgENicRDHSjmApGADO8zuAgA7XqQQhgAHRCTaMVVHl742ANp75tAJ4C+hVbQC0kdYVIubw48IBOUHoAuhxc7hoAYhD+xMK0WgBW8mrKsMxKiLqq2NjWdg5OtGZ5IHzxMDZ6mDo5pTb2MI7Obmoa3rS+UAFBCKXlhlUgtRxsQA):

<img width="254" alt="Screenshot 2024-08-25 at 16 43 06" src="https://github.com/user-attachments/assets/3d390df5-e0ab-4ad7-bc1b-a9cc9710f847">

Reply:
So there are two fixes in this PR.
You are right.